rust/automerge/src/op_tree.rs

```diff
@@ -278,46 +519,43 @@ impl PartialEq for OpTreeInternal {
     }
 }
 
-impl<'a> IntoIterator for &'a OpTreeInternal {
-    type Item = &'a Op;
-
-    type IntoIter = Iter<'a>;
-
-    fn into_iter(self) -> Self::IntoIter {
-        Iter {
-            inner: self,
-            index: 0,
-        }
-    }
+#[derive(Default, Debug, Clone, PartialEq)]
+pub(crate) struct OpsFound<'a> {
+    pub(crate) ops: Vec<&'a Op>,
+    pub(crate) ops_pos: Vec<usize>,
+    pub(crate) end_pos: usize,
 }
 
-pub(crate) struct Iter<'a> {
-    inner: &'a OpTreeInternal,
-    index: usize,
-}
-
-impl<'a> Iterator for Iter<'a> {
-    type Item = &'a Op;
-
-    fn next(&mut self) -> Option<Self::Item> {
-        self.index += 1;
-        self.inner.get(self.index - 1)
-    }
-
-    fn nth(&mut self, n: usize) -> Option<Self::Item> {
-        self.index += n + 1;
-        self.inner.get(self.index - 1)
+impl<'a> OpsFound<'a> {
+    fn new<T: Iterator<Item = &'a Op>>(
+        start_pos: usize,
+        mut iter: T,
+        key: Key,
+        clock: Option<&Clock>,
+    ) -> Self {
+        let mut found = Self {
+            end_pos: start_pos,
+            ops: vec![],
+            ops_pos: vec![],
+        };
+        let mut op = iter.nth(start_pos);
```

Comment:
Rather then using `nth` and then `loop` can we use `Iterator::skip(start_pos - 1)`?

Reply:
Because then we have to call `Iterator::next()` one more time than we need to in the absolute hottest function in the entire codebase? 😄

Reply:
Also start_pos could be zero